topic_tools/src/throttle_node.cpp

```diff
@@ -55,6 +55,11 @@ ThrottleNode::ThrottleNode(const rclcpp::NodeOptions & options)
 
 void ThrottleNode::process_message(std::shared_ptr<rclcpp::SerializedMessage> msg)
 {
+  std::scoped_lock lock(pub_mutex_);
```

Comment:
This lock was added in #49, but in practice it's not really needed since the `topic_tools` nodes all use the single-threaded executor by default